File: src/DynamicLinkLibraries/3DConverters/Abstract3DConverters/Interfaces/IMeshCreator.cs
using Abstract3DConverters.Materials;

namespace Abstract3DConverters.Interfaces
{
    /// <summary>
    /// Creator of meshes
    /// </summary>
    public interface IMeshCreator
    {
        /// <summary>
        /// Directory
        /// </summary>
        string Directory { get; }

        /// <summary>
        /// Loads itself
        /// </summary>
        /// <param name="bytes">Bytes</param>
        void Load(byte[] bytes);

        /// <summary>
        /// Meshes
        /// </summary>
        IEnumerable<IMesh> Meshes { get; }

        /// <summary>
        /// Effects
        /// </summary>
        Dictionary<string, Effect> Effects { get; }

        /// <summary>
        /// Name of file
        /// </summary>
        string Filename { get; }
    }
}

File: src/TypeScript/TypeScriptBackup/src/Library/Abstract3DConverters/Interfaces/IMeshCreator.ts
import type { IFactory } from "../../Interfaces/IFactory";
import type { IObject } from "../../Interfaces/IObject";
import type { IUrlObject } from "../../IO/Interfaces/IUrlObject";
import type { IMesh } from "./IMesh";
import { EffectTexture } from "../EffectTexture";


export interface IMeshCreator extends IObject, IUrlObject {


    /// <summary>
    /// Loads itself
    /// </summary>
    /// <param name="obj">Object</param>
    loadMeshCreator(): void;

    /// <summary>
    /// Meshes
    /// </summary>
    getMeshCreatorMeshes(): IMesh[]

    /// <summary>
    /// Effects
    /// </summary>
    getMeshCreatorEffects(): Map<string, EffectTexture>

    ///Gets the facrory
    getMeshCreatorFactory(): IFactory

    getMeshCreatorGenerator(): any

    getMeshCreatorDirectory(): string

}
